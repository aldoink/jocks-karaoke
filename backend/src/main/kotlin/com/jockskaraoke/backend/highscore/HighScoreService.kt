package com.jockskaraoke.backend.highscore

import org.springframework.stereotype.Service
import javax.transaction.Transactional

@Service
@Transactional
class HighScoreService(
    private val highScoreRepository: HighScoreRepository
) {
    fun findHighscores(songId: Int): List<HighScore> {
        return highScoreRepository.findAllBySongId(songId)
    }

    fun saveHighScore(name: String, score: Int, songId: Int) {
        highScoreRepository.save(HighScore(0, name, score, songId))
    }

    fun deleteHighScore(highScoreId: Int) {
        highScoreRepository.deleteById(highScoreId)
    }
}