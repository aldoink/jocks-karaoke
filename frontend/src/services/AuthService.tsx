import axios from "axios";

export const BACKEND_URL = process.env.REACT_APP_BACKEND_URL;

export class AuthService {
    async login (email: string, password: string): Promise<void> {
        const response = await axios.post(`${BACKEND_URL}/auth/login`, {email, password});
        localStorage.setItem("token", response?.data?.token);
    }

    getToken() {
        return localStorage.getItem("token")
    }
}