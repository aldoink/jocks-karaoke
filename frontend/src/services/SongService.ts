import axios from "axios";

export class SongService {
    private baseUrl = process.env.REACT_APP_BACKEND_URL;

    public async search(searchTerm: string) {
        let requestUrl = `${this.baseUrl}/songs`;
        if (searchTerm) {
            requestUrl += `?searchTerm=${searchTerm}`;
        }
        const response = await axios.get(requestUrl);
        return response?.data;
    }
}