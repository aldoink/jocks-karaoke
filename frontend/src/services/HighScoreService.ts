import axios, {AxiosRequestConfig} from "axios";
import {AuthService, BACKEND_URL} from "./AuthService";

export interface HighScore {
    [index: string]: string | number | undefined;

    name?: string;
    score?: string;
    songId: number;
}

export class HighScoreService {
    constructor(private authService: AuthService) {
    }

    public async findAll(songId: number): Promise<HighScore[]> {
        const response = await axios.get(`${BACKEND_URL}/highscores/${songId}`, this.getAxiosConfig());
        return response?.data;
    }

    public async save(entry: HighScore) {
        return await axios.put(
            `${BACKEND_URL}/highscores`,
            entry,
            this.getAxiosConfig()
        )
    }

    private getAxiosConfig(): AxiosRequestConfig {
        return {headers: {Authorization: "Bearer " + this.authService.getToken()}};
    }
}