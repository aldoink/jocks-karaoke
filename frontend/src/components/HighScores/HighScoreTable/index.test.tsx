import "jest-styled-components";
import { render } from "@testing-library/react";
import { HighScoreTable } from "./index";
import { HighScore } from "../../../services/HighScoreService";
import {
  HighScoreContext,
  IHighScoreContext,
} from "../../../contexts/HighScoreContext";

describe("HighScoreTable", () => {
  it("renders correctly", () => {
    const highScores = [
      { name: "Ally", score: "99", songId: 123 } as HighScore,
      { name: "Johnny", score: "98", songId: 123 } as HighScore,
      { name: "Jenny", score: "97", songId: 123 } as HighScore,
    ];

    expect(
      render(
        <HighScoreContext.Provider
          value={{ highScores } as unknown as IHighScoreContext}
        >
          {" "}
          <HighScoreTable />
        </HighScoreContext.Provider>
      ).asFragment()
    ).toMatchSnapshot();
  });
});
