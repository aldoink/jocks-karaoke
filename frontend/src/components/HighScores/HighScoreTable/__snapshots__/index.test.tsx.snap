// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighScoreTable renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  border-collapse: collapse;
}

.c1 td:first-child {
  width: 75%;
  margin-right: 1rem;
}

.c1 td CenteredColumn {
  width: 25%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<table
    class="c0"
  >
    <thead>
      <tr>
        <th
          class="name-column"
        >
          Name
        </th>
        <th
          class="score-column"
        >
          Score
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="c1"
      >
        <td>
          Ally
        </td>
        <td
          class="c2"
        >
          99
        </td>
      </tr>
      <tr
        class="c1"
      >
        <td>
          Johnny
        </td>
        <td
          class="c2"
        >
          98
        </td>
      </tr>
      <tr
        class="c1"
      >
        <td>
          Jenny
        </td>
        <td
          class="c2"
        >
          97
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
